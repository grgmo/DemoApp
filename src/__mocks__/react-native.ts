import RN from 'react-native';

RN.NativeModules.AppInfo = {
  getConstants: () => ({
    appVersion: '1.0.0',
  }),
};
module.exports = RN;
