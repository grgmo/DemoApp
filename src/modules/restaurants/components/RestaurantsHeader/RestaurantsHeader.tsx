import React, {FC} from 'react';
import {StyleSheet, View} from 'react-native';
import AppText from '../../../../common/components/AppText/AppText';
import theme from '../../../../common/theme';

type RestaurantsHeaderProps = {
  title: string;
  subTitle: string;
};

const RestaurantsHeader: FC<RestaurantsHeaderProps> = ({title, subTitle}) => {
  return (
    <View style={styles.header}>
      <AppText type="h1" text={title} bold style={styles.title} />
      <AppText type="h4" text={`Version ${subTitle}`} />
    </View>
  );
};

const styles = StyleSheet.create({
  header: {
    paddingVertical: theme.SPACE.MEDIUM,
    alignItems: 'center',
  },
  title: {
    marginBottom: theme.SPACE.XSMALL,
  },
});

export default RestaurantsHeader;
