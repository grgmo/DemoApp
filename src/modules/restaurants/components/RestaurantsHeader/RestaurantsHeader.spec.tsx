import React from 'react';
import {render} from '@testing-library/react-native';

import RestaurantsHeader from './RestaurantsHeader';

describe('RestaurantsHeader', () => {
  it('should render header correctly', () => {
    const header = render(
      <RestaurantsHeader title="title" subTitle="subTitle" />,
    );
    expect(header.toJSON()).toMatchSnapshot();
  });
});
