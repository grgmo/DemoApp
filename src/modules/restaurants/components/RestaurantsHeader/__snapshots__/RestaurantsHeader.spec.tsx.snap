// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantsHeader should render header correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "paddingVertical": 15,
    }
  }
>
  <Text
    accessibilityRole="text"
    style={
      Array [
        Object {
          "color": "black",
          "fontSize": 14,
        },
        Object {
          "marginBottom": 5,
        },
        Object {
          "fontSize": 22,
        },
        Object {
          "fontWeight": "bold",
        },
      ]
    }
  >
    title
  </Text>
  <Text
    accessibilityRole="text"
    style={
      Array [
        Object {
          "color": "black",
          "fontSize": 14,
        },
        Object {
          "fontSize": 14,
        },
      ]
    }
  >
    Version subTitle
  </Text>
</View>
`;
